viewing_party/party.py

```diff
@@ -1,23 +1,180 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    
+    if title and genre and rating:
+        new_movie = {}
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    else:
+        return None
+
+def add_to_watched(user_data, movie):
+
+    user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, movie_title):
+    for index in range(0, len(user_data["watchlist"])):
+        nested_movie_title_watched = user_data["watchlist"][index]["title"]
+        if movie_title in nested_movie_title_watched:
+            correct_movie = user_data["watchlist"][index]
+            user_data["watched"].append(correct_movie)
+            user_data["watchlist"].remove(correct_movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+
+    if user_data["watched"]:
+        rating_list = []
+        for index in range(0, len(user_data["watched"])):
+            rating= user_data["watched"][index]["rating"]
+            rating_list.append(rating)
+            average = sum(rating_list)/ len(rating_list)
+        return average
+    else:
+        return 0.0

```

Comment:
The `sum` method has a for loop under the hood, so let's combine them! Also, the variables of the for loop aren't very descriptive, which makes it hard for other coders to understand what you are comparing.

```suggestion
    if not user_data["watched"]:
        return 0.0
    rating_sum = 0
    for movie in user_data["watched"]:
        rating_sum += movie["rating"]
    return rating_sum/len(rating_list)
```